common/scala/src/main/scala/whisk/core/connector/Message.scala

```diff
@@ -53,7 +54,8 @@ case class ActivationMessage(
     activationId: ActivationId,
     activationNamespace: EntityPath,
     content: Option[JsObject],
-    cause: Option[ActivationId] = None)
+    cause: Option[ActivationId] = None,
+    traceMetadata: Option[SpanMetadata] = None)
```

Comment:
> Since you trace the request from controller to invoker, why do you not also trace the response back (via active ack) for blocking requests through the controller on the round trip?

@rabbah We don't need to trace response back. Because parent Trace is created in controller  we don't need to send back any trace context. `TransactionId` is used to fetch it from Map. Each span within a trace starts and ends when its individual `TransactionId` started and finished method calls. So for blocking request when active ack is received the parent TransactionId finishes and marks its span complete.